src/js/utils/FormUtil.js

```diff
@@ -0,0 +1,89 @@
+function cloneDefinition(array) {
+  return array.map(function (object) {
+    if (Array.isArray(object)) {
+      return object.slice();
+    }
+
+    if (typeof object === 'object') {
+      return Object.assign({}, object);
+    }
+
+    return object;
+  });
+}
+
+function getProp(key) {
+  return key.split('[')[0];
+}
+
+function getPropIndex(key) {
+  let value = key.split('[');
+  return parseInt(value[1].split(']')[0])
```

Comment:
As long as the regex is tested and with a clear, human readable description (or maybe even a comment) then I agree.